### PR TITLE
WIP: refactor chatstore and fix daymarkers

### PR DESCRIPTION
**Pull request by Simon-Laux**
- refactor chat store to use jsonrpc's messageGetMessages
- fix daymarkers
- split chat store into multiple files

needs https://github.com/deltachat/deltachat-core-rust/pull/3653 merged

Edit:
moved the testing points to #2948, will merge this pr now

### CODE REVIEW

File: src/renderer/components/message/MessageList.tsx [hunk 1]
```diff
@@ -147,7 +147,7 @@ export default function MessageList({
       if (!messageListRef.current) {
         return
       }
-      if (ChatStore.lockIsLocked('scroll') === true) {
+      if (ChatStore.sceduler.isLocked('scroll') === true) {
```
Comment (obsoleszenz) [inline src/renderer/components/message/MessageList.tsx:150]: scheduler

Comment (Jikstra) [inline src/renderer/components/message/MessageList.tsx:150]: scheduler

File: src/renderer/components/message/MessageList.tsx [hunk 2]
```diff
@@ -68,9 +69,8 @@ export default function MessageList({
   const {
     oldestFetchedMessageIndex,
     messagePages,
-    messageIds,
-    scrollTo,
-    lastKnownScrollHeight,
+    messageListItems,
+    viewState,
```
Comment (Jikstra) [inline src/renderer/components/message/MessageList.tsx:73]: nice!

File: src/renderer/stores/chat/chat_scroll.ts [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+  scrollTop: number
+}
+
+interface ScrollToBottom {
+  type: 'scrollToBottom'
+  /** toggle proximity check, if on scroll only if close */
+  ifClose: boolean
+}
+
+export class ChatViewState {
```
Comment (Jikstra) [inline src/renderer/stores/chat/chat_scroll.ts:32]: Is it really a state? It's more signaling something to the view

File: src/renderer/stores/chat/chat_locks.ts [hunk 4]
```diff
@@ -0,0 +1,193 @@
+import { getLogger } from '../../../shared/logger'
```
Comment (Jikstra) [inline src/renderer/stores/chat/chat_locks.ts:1]: I think chat_scheduler is a better name then chat_locks

File: src/renderer/stores/chat.ts [hunk 5]
```diff
@@ -10,7 +10,7 @@ import { debouncedUpdateBadgeCounter } from '../system-integration/badge-counter
 import { clearNotificationsForChat } from '../system-integration/notifications'
 import { T } from '@deltachat/jsonrpc-client'
 import { ChatViewState } from './chat/chat_scroll'
-import { ChatStoreSceduler } from './chat/chat_locks'
+import { ChatStoreScheduler as ChatStoreScheduler } from './chat/chat_locks'
```
Comment (obsoleszenz) [inline src/renderer/stores/chat.ts:13]:
```suggestion
import { ChatStoreScheduler } from './chat/chat_locks'
```

**Pull request merged**